gateway.js

```diff
@@ -0,0 +1,184 @@
+  });
+
+  gulp.task('clean-remote', 'clean remote', function(cb) {
+    all.sshExecCmd('sudo rm -rf ' + workspace, {
+      verbose: false
+    }, function(err) {
+      if (err) {
+        cb(err);
+      } else {
+        cb();
+      }
+    });
+  });
+
+  gulp.task('clean-local', 'clean local', function(cb) {
+    try {
+      if (all.fileExistsSync(config.bleConfig)) {
+        fs.unlinkSync(config.bleConfig);
+      }
+      if (all.fileExistsSync(config.azFuncConfig)) {
+        fs.unlinkSync(config.azFuncConfig);
+      }
+    } catch (err) {
+      cb(err);
+      return;
+    }
+    cb();
+  });
+
+  // remove the file in the local profile folder and remote NUC
+  gulp.task('clean', 'clean local and remote', ['clean-local', 'clean-remote']);
+
+  // discovery the ble devices
+  gulp.task('devdisco', 'discovery Sensortag device', function(cb) {
+    all.sshExecCmd('cd ' + workspace + '; node sensortagdisco.js', {
+      verbose: true
+    }, function(err) {
+      if (err) {
+        cb(err);
+      } else {
+        cb();
+      }
+    });
+  });
+
+  // test mac's connectivity
+  // usage: gulp testconnect --mac <mac address>
+  gulp.task('testconnect', 'test connectivity of mac address', function(cb) {
+    all.sshExecCmd('cd ' + workspace + '; node testconnect.js ' + args['mac'], {
+      verbose: true
+    }, function(err) {
+      if (err) {
+        cb(err);
+      } else {
+        cb();
+      }
+    });
+  });
+
+  // run samples on NUC
+  gulp.task('run', 'run ble_sample on NUC', ['setup-remote', 'upload-config'], function(cb) {
+    all.sshExecCmd('cd ' + workspace + '; node run.js', {
+      verbose: true
+    }, function(err) {
+      if (err) {
+        cb(err);
+      } else {
+        cb();
+      }
+    });
+  });
+
+  // copy all need file into NUC, and generate the ble_gateway.json
+  gulp.task('deploy', 'deplo ble_sample on NUC', ['setup-remote', 'upload-config'], function(cb) {
+    var force = args['force'] || args['f'];
+    var global = args['global'] || args['g'];
+    var options = '' + (force ? ' --force' : '') + (global ? ' --global' : '');
+    all.sshExecCmd('cd ' + workspace + '; node deploy.js' + options, {
+      verbose: true
+    }, function(err) {
+      if (err) {
+        cb(err);
+      } else {
+        cb();
+      }
+    });
+  });
+
+  // copy file into NUC
+  gulp.task('upload-config', 'upload config.json to NUC', function(cb) {
+    all.uploadFilesViaScp([getConfigFilepath(config.bleConfig)], [workspace + 'config.json'], cb);
+  });
+}
+
+function saveConfigFile(filename, config) {
+  var oldConfig = readConfig(filename);
+  var newConfig = Object.assign(config, oldConfig);
+  fs.writeFileSync(filename, JSON.stringify(newConfig, null, 2));
+}
+
+function getConfigFilepath(postfix) {
+  return all.getToolsFolder() + '/config-' + postfix + '.json';
+}
+
+function readConfig(filename) {
+  if (all.fileExistsSync(filename)) {
+    return require(filename);
+  }
+
+  return {};
+}
+
```

Comment:
I think we could move this to all.js

Currently all.js has function readCombinedConfig() with one parameter, it could accept array of prefixes, and flatten all given config files. This way we would have more generic function in all. and much less code here.

No need to do it in this pull request, but I think it should be done later.